debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -139,110 +150,213 @@ public String getName() {
      * @return the value; may be null
      */
     public Object getValue(String columnName, String columnType, Value rawValue, final PgConnectionSupplier connection) {
-        switch (columnType) {
+        // we support wal2json both before & after commit 0255f2ac, which means we may get old-style type names
+        // (eg. int2, timetz, varchar, numeric, _int4)
+        // or new-style/verbose ones
+        // (eg. smallint; time with time zone; character varying (255); decimal (10,3), integer[])
+
+        Matcher m = TYPE_PATTERN.matcher(columnType);
+        if (!m.matches()) {
+            LOGGER.error("Failed to parse columnType for {} '{}'", columnName, columnType);
+            throw new ConnectException(String.format("Failed to parse columnType '%s' for column %s", columnType, columnName));
+        }
+        String fullType = m.group("full");
+        String baseType = m.group("base").trim();
+        if (!Objects.toString(m.group("suffix"), "").isEmpty()) {
+            baseType = String.join(" ", baseType, m.group("suffix").trim());
+        }
+        String[] typeModifiers = {};
```

Comment:
Will this create a new array instance for each invocation or will be there only one shared?

Reply:
_`<not a java expert>`_ 
AFAIK the empty array is static & shared. If there are any modifiers found, then the `typeModifiers` reference is pointed to the new array from the regex 
_`</not a java expert>`_

Reply:
I tried this
```java
public class Test {

    @org.junit.Test
    public void test() {
        emptyArray();
        emptyArray();
        emptyArray();
    }

    private void emptyArray() {
        final String[] typeModifiers = {};
        System.out.println(typeModifiers);
    }
}
```
and got a unique value for each call

Reply:
So... how do I fix it?

Reply:
I think it is goin to be easy - just declare `private static final String[]` and use it as the empty array returned.